src/main/java/bio/terra/service/filedata/flight/ingest/BucketNameStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.DataLocationSelector;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+
+public class BucketNameStep implements Step {
+
+    private final UUID datasetId;
+    private final DataLocationSelector dataLocationSelector;
+
+    public BucketNameStep(UUID datasetId, DataLocationSelector dataLocationSelector) {
+        this.datasetId = datasetId;
+        this.dataLocationSelector = dataLocationSelector;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext flightContext) {
+        FlightMap workingMap = flightContext.getWorkingMap();
+        bio.terra.model.BillingProfileModel billingProfile =
+            workingMap.get(ProfileMapKeys.PROFILE_MODEL, bio.terra.model.BillingProfileModel.class);
+        String bucketName = dataLocationSelector.bucketForFile(datasetId, billingProfile);
+        workingMap.put(FileMapKeys.BUCKET_NAME, bucketName);
```

Comment:
If I'm understanding correctly, this is really only the potential bucket name. If a bucket already exists for a given dataset/billing profile combo, then the existing bucket name will be used again. The presence of a variable in the working map named "bucket name" that is not necessarily the bucket name could be confusing to future developers.

I could see a couple of ways to mitigate this:
1. Rename this variable to "POSSIBLE_BUCKET_NAME", thus making it clear to a user grabbing it downstream that it's not necessarily the correct bucket name
2. Break the logic up in the ResourceService so that you could actually pull the correct bucket name in this step (I'll add another comment there to show what I mean) and then just call getOrCreateBucket in the next step. I think this could be a bit cleaner solution & shouldn't be too hard, it's just re-arranging the code.